SwiftKitCore/src/main/java/org/swift/swiftkit/core/SwiftAsync.java

```diff
@@ -0,0 +1,31 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift.org project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+package org.swift.swiftkit.core;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public final class SwiftAsync {
+
+    public static final ExecutorService SWIFT_ASYNC_EXECUTOR = new ForkJoinPool(
+            Runtime.getRuntime().availableProcessors(),
+            ForkJoinPool.defaultForkJoinWorkerThreadFactory,
+            null,
+            true
+    );
```

Comment:
Yeah so with the blocking the current approach is doing, this becomes  tricky tradeoff. The FJP _will_ just keep adding threads (here's a fun [diagram](https://doc.akka.io/libraries/akka-http/current/handling-blocking-operations-in-akka-http-routes.html) i did ages ago and like to point out).

A ThreadPoolExecutor has fixed actually upholds the max thread count, so may be preferable for blocking work, at least on server -- idk how much folks on android care, but if we are trying to fix the number of blocked threads, the ThreadPoolExecutor would be better here...

_Having that said_, how about we explore the upcall+complete based implementation first? If it is acceptable we would not have to do any blocking which would be much preferable...